file/src/test/java/docs/javadsl/FileTailSourceTest.java

```diff
@@ -112,6 +118,116 @@ public void willReadNewLinesAppendedAfterReadingTheInitialContents() throws Exce
     subscriber.expectComplete();
   }
 
+  // #shutdown-on-delete
+
+  private static class Tick {}
+
+  /**
+   * Periodically checks if the file at `path` exists every `pollingInterval`. Shutdown the stream
+   * when the file is not found using an empty Source.
+   * @param pollingInterval Interval to check if file exists.
+   * @param path a file to check
+   */
+  private static Source<String, Cancellable> fileCheckSource(java.time.Duration pollingInterval, Path path) {
+    return Source
+            .tick(pollingInterval, pollingInterval, new Tick())
```

Comment:
You do not want to create a new instance for every tick. Create a static field with the instance.